### ISSUE TITLE
sample() method

### ISSUE THREAD

**Issue by MARALB83** [InvestmentSystems/static-frame#288]
It would be useful to add the pandas 'sample()' method to Frame. 

This method would be relevant to take a subset of a large dataset for exploratory purposes.  

Another use case might be for testing: to create a sample input case from some pre-generated table that is too large. 

With respect to the Pandas 0.23.4 interface of 'sample()', I think those arguments should be preserved. Most of the cases I ran into, just use 'n' or 'frac' coupled with 'axis'. I think it is very important to keep 'random_state' to be able to replicate prior runs. It is nice to have the optionality of 'replace' and 'weights'.

**Comment by MARALB83** [comment]
With regards to 'n' and 'frac', I can see how they might be merged into 1 argument, but I prefer to keep them separated to force the user to be explicit, and to be easier to read the code.

**Comment by flexatone** [comment]
Many thanks for this suggestion.

I support adding this functionality to all containers and should be able to get it into the next SF release.

In regard to the interface:

Pandas exposes both `n` and `frac` arguments, but only one can be supplied. This sort of interface, where arguments are dependent on each other and/or potentially in conflict, is something we try to avoid in SF. As a `n` can always be derived from `frac` by the caller (i.e., `frac * len(f.index)`), I prefer to only support an `n`-like interface.

Do you ever use the `weights` argument? This is a feature we might choose to defer, as I think most use-cases are fine with equal-weighted sampling. Adding it is easy, however, as it is part of the `np.random.choice` interface.

The `replace=True` cannot be supported in SF, as it would lead to a non-unique index on an axis, which is not permitted in SF.

Finally, rather than expose an `axis` argument, we might instead provide a `columns` and `index` argument for `Frame`, where the value is the count and both axis can be sampled  with one call.

So we might start with signatures like the following:

```python
Series.sample(count: int, seed: tp.Optional[int] = None)
Frame.sample(index: tp.Optional[int] = None, columns: tp.Optional[int] = None, seed: tp.Optional[int] = None)
```

What do you think?

**Comment by MARALB83** [comment]
In context of sampling universe or portfolio tables, I never came up with the need to use the `weights` argument. If we can always get it later, then it is ok to defer it.

Everything else are great improvements that make its usage clearer and easier.

**Comment by flexatone** [comment]
The sample() method is complete and will be available as part of the 0.7.8 release.  Thanks again for this suggestion.

```python
>>> f = sf.Frame(np.arange(20).reshape(4, 5))
>>> f
<Frame>
<Index> 0       1       2       3       4       <int64>
<Index>
0       0       1       2       3       4
1       5       6       7       8       9
2       10      11      12      13      14
3       15      16      17      18      19
<int64> <int64> <int64> <int64> <int64> <int64>

>>> f.sample(2, 3)
<Frame>
<Index> 2       3       4       <int64>
<Index>
2       12      13      14
3       17      18      19
<int64> <int64> <int64> <int64>
```
